src/realm/object-store/sync/sync_session.cpp

```diff
@@ -377,7 +390,7 @@ void SyncSession::handle_fresh_realm_downloaded(DBRef db, util::Optional<std::st
     // that moving to the inactive state doesn't clear them - they will be
     // re-registered when the session becomes active again.
     {
-        m_force_client_reset = true;
+        m_force_client_reset = allowed_mode ? allowed_mode : SyncError::ClientResetModeAllowed::RecoveryPermitted;
```

Comment:
i think you can do `allowed_mode.value_or(SyncError::ClientResetModeAllowed::RecoveryPermitted)`